jquery.ajaxchimp.js

```diff
@@ -93,15 +94,31 @@ For e.g. 'http://blahblah.us1.list-manage.com/subscribe/post-json?u=5afsdhfuhdsi
                         }
                     }
 
+
                     // Translate and display message
+                    var length = 0, msgnr;
+
+                    if(Object.keys){
+                        length = Object.keys($.ajaxChimp.responses).length
+                    }
+                    else {
+                        for (var key in $.ajaxChimp.responses){
+                            if ($.ajaxChimp.responses.hasOwnProperty(key)) {
+                              length++;
+                            }
+                        }
+                    }
+
+                    while(length--) {
+                        if (msg.indexOf($.ajaxChimp.responses[length])!==-1) {
+                            msgnr = length;
+                        }
+                    }
```

Comment:
I did a similar fix for this locally, but I didn't change the responses structure.
My fix looks like:
```js
var i18n = $.ajaxChimp.translations;
// Translate and display message
if (settings.language !== 'en' && i18n && i18n[settings.language]) {
    var translation = i18n[settings.language][$.ajaxChimp.responses[msg]];
    if (!translation) {
        // try partial match
        $.each($.ajaxChimp.responses, function(resp, key) {
            if (msg.indexOf(resp) > -1) {
                translation = i18n[settings.language][key];
                return;
            }
        });
    }
    if (translation) {
        msg = translation;
    }
}
```